client/gutenberg/editor/controller.js

```diff
@@ -35,6 +41,36 @@ function getPostID( context ) {
 	return parseInt( context.params.post, 10 );
 }
 
+export const jetpackBlocki18n = ( context, next ) => {
+	if ( ! isEnabled( 'gutenberg/block/jetpack-preset' ) ) {
+		return next();
+	}
+
+	const state = context.store.getState();
+	const localeSlug = getCurrentLocaleSlug( state );
+
+	// We don't need to localize English
+	if ( localeSlug === 'en' ) {
+		return next();
+	}
+
+	const languageFileUrl =
+		'https://widgets.wp.com/languages/jetpack-gutenberg-blocks/' + localeSlug + '.json';
+
+	request.get( languageFileUrl ).end( ( error, response ) => {
```

Comment:
we shouldn't need to pull in `superagent` here and instead could consider using the data layer…

```js
import { requestHttpData, waitForData } from 'state/data-layer/http-data';
import { rawHttp } from 'state/data-getters';

waitForData( {
	translations: requestHttpData(
		'jetpack-gutenberg-translations',
		rawHttp( {
			method: 'GET',
			url: `https://widgets.wp.com/languages/jetpack-gutenberg-blocks/${ localeSlug }.json`,
		} ),
		{ fromApi: () => ( { body } ) => [ [ 'jetpack-gutenberg-translations', body ] ] }
	)
} ).then( ( { translations: { state, data } ) => {
	if ( state === 'failure' ) {
		debug( `Encountered an error loading locale file for ${ localeSlug }. Falling back to English.` );
	}

	// could still be expired data in the system
	if ( data ) {
		setLocaleData( data, 'jetpack' );
	}

	next();
} );
```

granted, that's more code, but it's inspectable in our dev tools and comes with freshness and auto-retry policies. if you are interested I'd be happy to create a helper function in `state/data-getters` which cuts down on the overhead for doing a basic raw HTTP request

```js
waitForData( {
	translations: http( 'GET', `https://widgets.wp.com/…` ),
} ).then( ( { translations: { state, data } } ) => {
	if ( 'failure' === state ) {
		debug( '…' );
	}

	if ( data ) {
		setLocaleData( data.body, 'jetpack' );
	}

	next();
} );
```

Reply:
https://github.com/Automattic/wp-calypso/pull/28867

^^^ in case you are interested…

Reply:
This is lovely, thanks @dmsnell ❤️ 

I've implemented your suggestion in this PR, and it's ready for another look.